filebeat/test_filebeat.py

```diff
@@ -0,0 +1,94 @@
+# (C) Datadog, Inc. 2010-2016
+# All rights reserved
+# Licensed under Simplified BSD License (see LICENSE)
+
+# stdlib
+from nose.plugins.attrib import attr
+from collections import namedtuple
+import os
+
+# 3p
+from mock import patch
+
+# project
+from tests.checks.common import AgentCheckTest, Fixtures
+
+
+mocked_file_stats = namedtuple('mocked_file_stats', ['st_size', 'st_ino', 'st_dev'])
+
+
+# allows mocking `os.stat` only for certain paths; for all others it will call
+# the actual function - needed as a number of test helpers do make calls to it
+def with_mocked_os_stat(mocked_paths_and_stats):
```

Comment:
This is fine, but perhaps over-complicated? We're not really every going to stat a real file from what I see in the fixtures, right? Why not just use a full `Mock` and pass it to patch as the `new_callable`. Maybe I'm missing, by all means let me know if I'm missing something.